src/EnergyPlus/CurveManager.hh

```diff
@@ -328,12 +364,16 @@ struct CurveManagerData : BaseGlobalStruct
     bool GetCurvesInputFlag = true;
     bool CurveValueMyBeginTimeStepFlag = false;
     bool FrictionFactorErrorHasOccurred = false;
-    Array1D<Curve::PerformanceCurveData> PerfCurve;
+    bool showFallbackMessage = true;
+    Array1D<Curve::Curve *> PerfCurve;
```

Comment:
I would like to do something here, maybe not in this PR. I want all objects that are stored as arrays of pointers to be: i) subtypes of an EPObject class that contains one field `std::string name` (or maybe a second field `std::string nameUC`), and ii) stored in a `std::vector` and for their indices to be 0 based, with -1 meaning not found. These two things are separate of course, but would be good to knock them both out. Will do this with `Material` also.

Reply:
I could easily move this to an EPVector as a step in the right direction.  I didn't change it to a std::vector because the index usage is spread so far around everywhere that I was worried about index issues causing lots of diffs or failures to resolve in the same PR as other big changes.  I am happy to move forward on these ideas though as we continue the refactor.

Reply:
I understand.